arborio/include/arborio/loaded_morphology.hpp

```diff
@@ -0,0 +1,75 @@
+#pragma once
+
+#include <variant>
+#include <any>
+
+#include <arbor/morph/label_dict.hpp>
+#include <arbor/morph/morphology.hpp>
+#include <arbor/morph/segment_tree.hpp>
+
+namespace arborio {
+
+struct ARB_SYMBOL_VISIBLE swc_metadata {};
+
+struct ARB_SYMBOL_VISIBLE asc_color {
+    uint8_t r = 0;
+    uint8_t g = 0;
+    uint8_t b = 0;
+};
+
+struct ARB_SYMBOL_VISIBLE asc_spine {
+    std::string name;
+    arb::mpoint location;
+};
+
+enum ARB_SYMBOL_VISIBLE asc_marker { dot, circle, cross, none };
+
+struct ARB_SYMBOL_VISIBLE asc_marker_set {
+    asc_color color;
+    asc_marker marker = asc_marker::none;
+    std::string name;
+    std::vector<arb::mpoint> locations;
+};
+
+struct ARB_SYMBOL_VISIBLE asc_metadata {
+    std::vector<asc_marker_set> markers;
+    std::vector<asc_spine> spines;
+};
+
+// Bundle some detailed metadata for neuroml ingestion.
+struct ARB_SYMBOL_VISIBLE nml_metadata {
```

Comment:
consider including `<arborio/export.hpp>` and use `ARB_ARBORIO_API` for "normal" classes which don't need rtti, see [docs](https://docs.arbor-sim.org/en/latest/dev/export.html).